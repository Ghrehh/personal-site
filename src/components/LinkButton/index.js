// @flow
import * as React from 'react';
import { Link } from 'react-router-dom';

import Button from 'components/Button';
import styles from './styles.module.css';

type Props = {
  className?: string,
  children?: React.Node,
  link: string
};

const scrollToTop = () => window.scrollTo(0, 0);

const LinkButton = (props: Props) => (
  <Link to={props.link} className={styles.link}>
    <Button onClick={scrollToTop} className={props.className}>
        {props.children}
    </Button>
  </Link>
);

export default LinkButton;
