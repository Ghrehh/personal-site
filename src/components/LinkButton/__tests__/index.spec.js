import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import LinkButton from '../';

jest.mock('react-router-dom', () => ({ Link: 'div' }));

describe('LinkButton', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};

    testContext.component = <LinkButton link="./page">Foo</LinkButton>;
  });

  describe('rendering', () => {
    beforeEach(() => {
      testContext.renderer = new ShallowRenderer();
      testContext.renderer.render(testContext.component);
      testContext.result = testContext.renderer.getRenderOutput();
    });

    it('renders the component correctly', () => {
      expect(testContext.result).toMatchSnapshot();
    });
  });
});
