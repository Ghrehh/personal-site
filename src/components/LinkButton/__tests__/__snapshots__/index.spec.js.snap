// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton rendering renders the component correctly 1`] = `
<LinkButton
  external={false}
  link="./page"
  scrollToTop={true}
>
  <Button
    className={undefined}
  >
    Foo
  </Button>
</LinkButton>
`;
